architecture/core_concepts/routes.adoc

```diff
@@ -12,147 +12,136 @@ toc::[]
-calls to another system, such as *F5*.
+A router uses the service selector to find the service and the endpoints
+backing the service. Service-provided load balancing is bypassed and
+replaced with the router's own load balancing. Routers watch the cluster
+API and automatically update their own configuration according to any
+relevant changes in the API objects. Routers may be containerized or
+virtual. Custom routers can be deployed to communicate modifications of
+API objects to another system, such as an *F5*.
+
+In order to reach a router in the first place, requests for hostnames
+must resolve via DNS to a router or set of routers. The suggested
+method is to define a cloud domain with a wildcard DNS entry pointing
+to a virtual IP backed by multiple router instances on designated nodes.
+link:../../admin_guide/high_availability.html#configuring-a-highly-available-routing-service[Router
+vIP configuration is described in the Administration Guide]. DNS
+for addresses outside the cloud domain would need to be configured
+individually. Other approaches may be feasible.
+
+=== Template Routers
+Currently, a template router is the single type of router plug-in
+available in OpenShift.
 
-Other capabilities exist to load-balance a service within a cluster. These
-services are exposed via a configurable link relation between different
-services, and ensure a set of services can be available.
-link:../../dev_guide/deployments.html[Deployments] can use these services as
-local proxies for each host, or reuse the shared routing infrastructure.
+A template router provides certain infrastructure information to the underlying
+router implementation, such as:
 
-As an OpenShift administrator, you can configure routers in your instance. This
-allows developers to then set the route types for their projects.
+- A wrapper that watches endpoints and routes.
+- Endpoint and route data, which is saved into a consumable form.
+- Passing the internal state to a configurable template and executing the
+template.
+- Calling a reload script.
 
-== Route Types
-Routes can be either secure or unsecure. Secure routes provide the ability to
-use different types of TLS termination to serve certificates to the client.
-Routers support link:#edge-termination[edge],
-link:#passthrough-termination[passthrough], and
-link:#re-encryption-termination[re-encryption] termination.
+Router plug-ins assume they can bind to host ports 80 and 443. This is to allow
+external traffic to route to the host and subsequently through the router.
+Routers also assume that networking is configured such that it can access all
+pods in the cluster.
 
-To create a secure route, specify the TLS termination of the route in a JSON
-file.
+[[haproxy-template-router]]
+=== HAProxy Template Router
+The HAProxy template router implementation is the reference implementation for a
+template router plug-in. It uses the *openshift/origin-haproxy-router*
```

Comment:
We need to conditionalize the image a la https://docs.openshift.com/enterprise/3.0/admin_guide/install/deploy_router.html#overview.